src/main/com/aca/homework/week13/race/condition/list/SafeList.java

```diff
@@ -0,0 +1,18 @@
+package com.aca.homework.week13.race.condition.list;
+
+import java.util.ArrayList;
+
+public class SafeList<E> implements List<E> {
+
+    private final java.util.List<E> list = new ArrayList<>();
+
+    @Override
+    public synchronized void add(E item) {
+        list.add(item);
+    }
+
+    @Override
+    public int size() {
```

Comment:
this method should be synchronized too. It will create happens-before relation. If you write using synchronized method, you should read the result via synchronized method too